### ISSUE TITLE
Model Summary not working 

### ISSUE THREAD

**Issue by JonathanBhimani-Burrows** [manujosephv/pytorch_tabular#348]
**Describe the bug**
When trying to print the model summary, receive error 
AttributeError: 'TabularModel' object has no attribute 'model'

**To Reproduce**

Set up tabular model (similar to the bank example in the repo) 

data_config = DataConfig(
    target=['xxx'], #target should always be a list.
    continuous_cols=num_cols_list,
    categorical_cols=cat_cols_list,
)

trainer_config = TrainerConfig(
auto_lr_find=True, # Runs the LRFinder to automatically derive a learning rate
    batch_size=256,
    max_epochs=500,
    early_stopping=None,
    checkpoints="valid_loss", # Save best checkpoint monitoring val_loss
    load_best=True, # After training, load the best checkpoint
)

optimizer_config = OptimizerConfig()

head_config = LinearHeadConfig(
    layers="", # No additional layer in head, just a mapping layer to output_dim
    dropout=0.1,
    initialization="kaiming"
).__dict__ # Convert to dict to pass to the model config (OmegaConf doesn't accept objects)

model_config = CategoryEmbeddingModelConfig(
    task="classification",
    layers="64-32",  # Number of nodes in each layer
    activation="ReLU", # Activation between each layers
    learning_rate = 1e-3,
    head = "LinearHead", #Linear Head
    head_config = head_config, # Linear Head Config

)

tabular_model = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config,
)

#try printing model 
print(tabular_model.summary(max_depth=-1))

**Expected behavior**
Model summary should appear 

**Screenshots**

**Desktop (please complete the following information):**

- OS:  Windows 11 (running on Google Collab)
- Browser : Edge 
- Version: 

**Smartphone (please complete the following information):**

- Device: \[e.g. iPhone6\]
- OS: \[e.g. iOS8.1\]
- Browser \[e.g. stock browser, safari\]
- Version 120.0.2210.77 (Official build) (64-bit)

**Additional context**

**Comment by manujosephv** [comment]
@JonathanBhimani-Burrows Good spot! 

So, the `TabularModel` does not initialize the underlying PyTorch model just yet. Because usually we need information like the number of input features etc to concretely initialize the model. So, PyTorch Tabular typically does that during fit. And unless that happens, the `summary` method isn't going to work. But, all this isn't clear from the error message.

So, I have merged a PR #350 which enables this behaviour..

After the PR, there are three different behaviours of the summary method:
```python
abular_model = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config,
)

tabular_model.summary() #1
dm = tabular_model.prepare_dataloader(train, test)
model = tabular_model.prepare_model(dm)
tabular_model.summary(model) #1
tabular_model.train(model, dm)
tabular_model.summary() #1
```

No. 1: When summary is called before fitting, it displays the model name, and selected config
No.2: If we pass in an initialized model to the summary as an argument, it will show the summary of the model
No.3: If we call `summary` after fitting or training the model, it shows the summary of the model which it was trained

**Comment by JonathanBhimani-Burrows** [comment]
Excellent, fantastic work! 
Closing the issue
